assets/js/components/DashboardNavigation.js

```diff
@@ -143,32 +166,112 @@ export default function DashboardNavigation() {
 		}
 	} );
 
+	useEffect( () => {
+		const onScroll = throttle( () => {
+			let closest;
+			let closestID = ANCHOR_ID_TRAFFIC;
+
+			for ( const areaID of [
+				ANCHOR_ID_TRAFFIC,
+				ANCHOR_ID_CONTENT,
+				ANCHOR_ID_SPEED,
+				ANCHOR_ID_MONETIZATION,
+			] ) {
+				const top = document
+					.getElementById( areaID )
+					.getBoundingClientRect().top;
+
+				/**
+				 * Gets the sticky elements - admin bar, header and navigation height into account.
+				 * And subtracts them with the top. When the user scrolls the page,
+				 * an appropriate chip becomes selected and the URL hash changes to the new anchor.
+				 */
+				const header = document.querySelector(
+					'.googlesitekit-header'
+				);
+
+				const hasStickyAdminBar = breakpoint !== 'small';
+
+				const headerHeight = hasStickyAdminBar
+					? header.getBoundingClientRect().bottom
+					: header.offsetHeight;
+
+				const navigation = document.querySelector(
+					'.googlesitekit-navigation'
+				);
+				const navigationHeight = navigation.offsetHeight;
+
+				const marginBottom = 20;
+				const topExcludingStickyElements =
+					top - headerHeight - navigationHeight - marginBottom;
+
+				if (
+					topExcludingStickyElements < 0 &&
+					( closest === undefined ||
+						closest < topExcludingStickyElements )
+				) {
+					closest = topExcludingStickyElements;
+					closestID = areaID;
+				}
+			}
+
+			const scrollOffset =
+				global.scrollY || document.documentElement.scrollTop;
+			const selectedElement = closest + scrollOffset;
+
+			/**
+			 * Calculates whether the screen is scrolled to the appropriate section.
+			 * And then sets the shouldScroll ref value to true.
+			 * This ensures setting the selectedID and updating the hash in the URL.
+			 */
+			if ( scrollOffset - selectedElement < 25 ) {
+				shouldScroll.current = true;
+			}
+
+			/**
+			 * This will be executed when the `scroll` event is triggerred
+			 * as well as the handleSelect callback is fired.
+			 */
+			if ( shouldScroll.current ) {
+				global.history.replaceState( {}, '', `#${ closestID }` );
+				setSelectedID( closestID );
+			}
+		}, 50 );
+
+		global.addEventListener( 'scroll', onScroll );
+
+		return () => {
+			global.removeEventListener( 'scroll', onScroll );
+		};
+	}, [ breakpoint ] );
+
 	return (
-		<ChipSet
-			className="googlesitekit-navigation"
-			selectedChipIds={ selectedIds }
-			handleSelect={ handleSelect }
-			choice
-		>
+		<div className="googlesitekit-navigation mdc-chip-set">
```

Comment:
Setting `mdc-chip-set` fixes the style issues on small and larger screens. We don't need to set `display: flex` for `googlesitekit-navigation`.